src/lib/components/legend/Legend.svelte

```diff
@@ -42,167 +33,96 @@
 		{ value: 'fundingSource', label: 'Funding' }
 	];
 
-	function handleModeChange(event: Event) {
-		const target = event.target as HTMLInputElement;
-		visualState.update(state => ({ 
-			...state, 
-			colorMode: target.value as typeof state.colorMode,
+	function handleModeChange(event: Event & { currentTarget: HTMLInputElement }) {
+		visualState.set({
+			mode: event.currentTarget.value as Mode,
 			filters: new Set()
-		}));
+		});
 	}
 
 	function toggleFilter(label: string) {
-		const newFilters = new Set($visualState.filters);
-		
-		if (newFilters.has(label)) {
-			newFilters.delete(label);
-		} else {
-			if (newFilters.size === 0) {
-				newFilters.add(label);
-			} else {
-				if (label === 'Other') {
-					newFilters.add('Other');
-				} else if (CATEGORIES[$visualState.colorMode].includes(label)) {
-					newFilters.delete('Other');
-					newFilters.add(label);
-				} else {
-					newFilters.add(label);
-				}
-			}
+		const updatedFilters = new Set($visualState.filters);
+
+		if (updatedFilters.has(label)) {
+			updatedFilters.delete(label);
+		} else if (label === 'Other') {
+			updatedFilters.add(label);
+		} else if (CATEGORIES[$visualState.mode].includes(label)) {
+			updatedFilters.delete('Other');
+			updatedFilters.add(label);
 		}
-		
-		const updatedFilters = new Set(newFilters);
-		visualState.update(state => ({
-			...state,
-			filters: updatedFilters
-		}));
-	}
 
-	$: isItemActive = (label: string) => $visualState.filters.has(label);
+		visualState.update((state) => {
+			state.filters = updatedFilters;
 
-	const longestLabel = Math.max(
-		...Object.values(legendItems).flatMap((items) => items.map((item) => item.label.length))
-	);
-
-	const minPanelWidth = longestLabel * 6 + 20;
+			return state;
+		});
+	}
 </script>
 
 <div
 	class={[
-		'floating-panel absolute z-[15] bg-white px-2 pb-2 pt-0.5 shadow-lg md:bottom-auto md:left-auto md:right-[calc(3%+48px)] md:top-4 md:block',
+		'floating-panel absolute z-[15] px-2 pb-2 pt-0.5 md:bottom-auto md:left-auto md:right-[calc(3%+48px)] md:top-4 md:block',
 		$uiState.legendExpanded ? 'bottom-[calc(40px+0.5rem)] left-[calc(3%+5rem)]' : 'hidden'
 	]}
-	style="min-width: {minPanelWidth}px"
 >
-	<div class="mb-0 flex items-center gap-1">
-		<svg
-			xmlns="http://www.w3.org/2000/svg"
-			viewBox="0 0 20 20"
-			fill="currentColor"
-			class="h-3 w-3 text-gray-500"
-		>
+	<div class="flex items-center gap-1 text-gray-500">
+		<svg height="20" width="20" viewBox="0 0 20 20" fill="currentColor" class="h-3 w-3">
 			<path
 				fill-rule="evenodd"
 				d="M2.628 1.601C5.028 1.206 7.49 1 10 1s4.973.206 7.372.601a.75.75 0 01.628.74v2.288a2.25 2.25 0 01-.659 1.59l-4.682 4.683a2.25 2.25 0 00-.659 1.59v3.037c0 .684-.31 1.33-.844 1.757l-1.937 1.55A.75.75 0 018 18.25v-5.757a2.25 2.25 0 00-.659-1.591L2.659 6.22A2.25 2.25 0 012 4.629V2.34a.75.75 0 01.628-.74z"
 				clip-rule="evenodd"
 			/>
 		</svg>
-		<p class="font-['Basis_Grotesque'] text-xs leading-tight text-gray-500">
```

Comment:
Rather than always adding in computed values for something like `font-family`, just add it to the `tailwind.config`! You'll notice I added `basis` and `poly` to `tailwind.config.ts`.